index.ts

```diff
@@ -0,0 +1,60 @@
+const products = require("./data/products.json");
+
+interface Product {
+  name: string;
+  description: string;
+  price: number;
+  image: URL;
+  limit: number;
+  amount: number;
+}
+
+interface Cart {
+  products: Product[];
+}
+
+const addProduct = (cart: Cart, product: Product) => {
+  cart.products.push(product);
+};
+
+const removeProduct = (cart: Cart, productToRemove: Product) => {
+  cart.products.forEach((product, i) => {
+    if (product.name === productToRemove.name) cart.products.splice(i, 1);
+  });
+};
+
+const updateProductAmount = (
+  cart: Cart,
+  productToChangeAmount: Product,
+  amount: number
+) => {
+  cart.products.forEach((product, i) => {
+    if (product.name === productToChangeAmount.name) product.amount = amount;
+  });
+};
+
+const checkout = (cart: Cart) => {
+  cart.products = [];
+};
+
+const totalPrice = (cart: Cart): number => {
+  let sum: number = 0;
+  cart.products.forEach((product, i) => {
```

Comment:
why do you need `i` ? when do we use `forEach` and when do we use `map`?

Reply:
fixed

Reply:
map returns a new array that his elements are all the returned values of the functions. foreach invokes all the functions (callback functions - one per element), like map, but doesn't return anything.